### PR TITLE
Flags -> LaneConnectionManager

### PR DESCRIPTION
**Pull request by Elesbaan70**
Remove all lane connection logic from the Flags class and set up a framework in LaneConnectionManager that can support connection flags and other new features.
see #1392

### CODE REVIEW

File: TLM/TLM/Manager/Impl/LaneConnectionManager.cs [hunk 1]
```diff
@@ -832,44 +871,42 @@ public class LaneConnectionManager
 
             for (uint i = 0; i < Singleton<NetManager>.instance.m_lanes.m_buffer.Length; i++) {
                 try {
-                    if (Flags.laneConnections[i] == null) {
-                        continue;
-                    }
+                    if (Flags.CheckLane(i)) {
```
[the remark this answers is not in the document]
Comment (Elesbaan70) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs:874]: That sounds possible, though I'm not where I can easily look at the code right now. But it's outside the scope of lane connections. Eliminating that method would affect a lot of other classes.

[the remark this answers is not in the document]
Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs:874]: if I am not mistaken check lane also removes invalid lanes too. In any case based on the discussion we had in discord this seems to be a duplicate of my PR.

**Comment by kianzarrin** [comment]
I converted this to draft because this is a duplicate of my PR.
If you don't intend to merge this code please close the PR.

File: TLM/TLM/Manager/Impl/LaneConnectionManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aubergine10) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs:874]: @kianzarrin No, it just checks the lane for a specific set of problems:

```csharp
        internal static bool CheckLane(uint laneId) {

            if (laneId <= 0) {
                return false;
            }

            ref NetLane netLane = ref laneId.ToLane();

            if (((NetLane.Flags)netLane.m_flags & (NetLane.Flags.Created | NetLane.Flags.Deleted)) != NetLane.Flags.Created) {
                return false;
            }

            ushort segmentId = netLane.m_segment;
            if (segmentId <= 0) {
                return false;
            }

            ref NetSegment netSegment = ref segmentId.ToSegment();

            return (netSegment.m_flags & (NetSegment.Flags.Created | NetSegment.Flags.Deleted)) == NetSegment.Flags.Created;
        }
```

The call site is responsible for removing any customisations should an invalid lane be detected.

**Comment by Elesbaan70** [comment]
It *could* be merged, so closing it seems unnecessary until another merge makes it obsolete. But making it Draft is the right thing to do, since I think even if it were to be used, it shouldn't be merged until the LaneConnection class has been taken out of the API. It is an implementation detail that should not be made public.

**Comment by Elesbaan70** [comment]
This has been abandoned in favor of a different approach to displaced lane support that does not directly involve LaneConnectionManager

**Pull request closed without merging**